test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -227,38 +225,41 @@ protected AggregationContext createAggregationContext(IndexSearcher indexSearche
          */
         BigArrays bigArrays = new MockBigArrays(new MockPageCacheRecycler(Settings.EMPTY), breakerService).withCircuitBreaking();
 
-        // TODO: now just needed for top_hits, this will need to be revised for other agg unit tests:
-        MapperService mapperService = mock(MapperService.class);
-        when(mapperService.getIndexSettings()).thenReturn(indexSettings);
-        when(mapperService.hasNested()).thenReturn(false);
-        when(mapperService.indexAnalyzer(anyString(), any())).thenReturn(Lucene.STANDARD_ANALYZER); // for significant text
-        for (MappedFieldType type : fieldTypes) {
-            String name = type.name();
-            when(mapperService.fieldType(name)).thenReturn(type);
-            // Alias each field to <name>-alias so everyone can test aliases
-            when(mapperService.fieldType(name + "-alias")).thenReturn(type);
-        }
-        when(mapperService.getObjectMapper(anyString())).thenAnswer(invocation -> {
-            String fieldName = (String) invocation.getArguments()[0];
-            if (fieldName.startsWith(NESTEDFIELD_PREFIX)) {
-                return new ObjectMapper.Builder(fieldName, Version.CURRENT).nested(Nested.newNested()).build(new ContentPath());
-            }
-            return null;
-        });
+        MappingLookup mappingLookup = new MappingLookup(
+            "_doc",
+            Arrays.stream(fieldTypes).map(MappingLookupUtils::mockFieldMapper).collect(toList()),
+            objectMappers(),
+            // Alias all fields to <name>-alias to test aliases
+            Arrays.stream(fieldTypes)
+                .map(ft -> new FieldAliasMapper(ft.name() + "-alias", ft.name() + "-alias", ft.name()))
+                .collect(toList()),
+            List.of(),
+            0,
+            souceToParse -> null,
+            true
+        );
 
         TriFunction<MappedFieldType, String, Supplier<SearchLookup>, IndexFieldData<?>> fieldDataBuilder = (
             fieldType,
             s,
-            searchLookup) -> fieldType.fielddataBuilder(mapperService.getIndexSettings().getIndex().getName(), searchLookup)
+            searchLookup) -> fieldType.fielddataBuilder(indexSettings.getIndex().getName(), searchLookup)
                 .build(new IndexFieldDataCache.None(), breakerService);
+        BitsetFilterCache bitsetFilterCache = new BitsetFilterCache(indexSettings, new BitsetFilterCache.Listener() {
```

Comment:
how bad would you feel about leaving this test as-is, and simply providing mapperService.mappingLookup() to the query shard context constructor? Would that work? I figured we'll get back to this anyways once we work on NestedDocuments access.

Reply:
I didn't mock `MappingLookup` at first because we were exposing `FieldTypeLookup` and it would have made mocks returning mocks which would have been a pain. So I ended up doing this to use the "real" stuff because it was just a ton of work to mock.

I guess at this point we don't need to keep them because we are routing everything through `MappingLookup` but they were a pain to get right so I'd like to keep them. But if you are really uncomfortable with them I could try and fork them out and get them back in.

Reply:
I couldn't keep the test as is though - I'd have to mock `MappingLookup` instead of `MapperService` i believe.